app/Models/PhotoManager.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Proto\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Auth;
+
+use Exception;
+
+class PhotoManager extends Model
+{
+    public static function getAlbums($max = null, $query=null, $unpublished=false, $no_thumb=true)
+    {
+        $include_private = (Auth::check() && Auth::user()->member() !== null);
+
+        $base = PhotoAlbum::orderBy('date_taken', 'desc');
+        if (!$include_private) {
+            $base = $base->where('private', '=', false);
+        }
+        if ($query) {
+            $base = $base->where('name', 'LIKE', '%'.$query.'%');
+        }
+        if ($unpublished) {
+            $base = $base->where('published', '=', false);
+        } else {
+            $base = $base->where('published', '=', true);
+        }
+        if (!$no_thumb) {
+            $base = $base->where('thumb_id', '!=', 'null');
+        }
+        if ($max != 0) {
+            $albums = $base->paginate($max);
+        } else {
+            $albums = $base->get();
+        }
+        return $albums;
+    }
+
+
+
+    public static function getPhotos($albumID)
+    {
+        $include_private = (Auth::check() && Auth::user()->member() !== null);
+
+        $album = PhotoAlbum::where('id', $albumID);
+        if (!$include_private) {
+            $album->where('private', '=', false);
+        }
+        $album = $album->get();
```

Comment:
I think $album->findOrFail will also work and obsolete the three lines below, but not sure.

Reply:
Different implementation chosen since throwing 404 errors in models is not nice